Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -16,13 +16,28 @@
 
 #include "Firestore/core/src/firebase/firestore/remote/datastore.h"
 
+#include <algorithm>
+
 #include "Firestore/core/include/firebase/firestore/firestore_errors.h"
 #include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
 
 namespace firebase {
 namespace firestore {
 namespace remote {
 
+namespace {
+
+absl::string_view ToStringView(grpc::string_ref grpc_str) {
```

Comment:
Creating an `absl::string_view` from an `NSString` is inefficient, because it's actually creating an autoreleased `NSData` buffer to back the view, which is less efficient than just copying into a `std::string`. Creating an `absl::string_view` from another kind of string view should be trivially cheap.

These cases are very different, and we're not looking to eliminate the creation of `absl::string_view` at all--just in places where we were thinking we wrapping `NSString`.